src/__internal__/focus-trap/focus-trap.spec.tsx

```diff
@@ -706,34 +684,16 @@ describe("FocusTrap", () => {
     it("tab should cycle through focusable elements inside the provided container refs and ignore all others", () => {
       focusElement(screen.getByText(BUTTON_IN_WRAPPER));
       tabPress();
-      expect(screen.getByText(BUTTON_IN_ADDITIONAL_WRAPPER_ONE)).toBeFocused();
+      expect(screen.getByText(BUTTON_IN_ADDITIONAL_WRAPPER_ONE)).toHaveFocus();
       tabPress();
-      expect(screen.getByText(BUTTON_IN_ADDITIONAL_WRAPPER_TWO)).toBeFocused();
+      expect(screen.getByText(BUTTON_IN_ADDITIONAL_WRAPPER_TWO)).toHaveFocus();
       tabPress();
-      expect(screen.getByText(BUTTON_IN_WRAPPER)).toBeFocused();
+      expect(screen.getByText(BUTTON_IN_WRAPPER)).toHaveFocus();
     });
   });
 
   describe("when content in the children tree changes", () => {
-    it("should trigger the MutationObserver", () => {
-      const mutationObserverMock = jest.fn(function MutationObserver(
-        this: {
-          observe: jest.Mock;
-          disconnect: jest.Mock;
-          trigger: () => void;
-        },
-        callback
-      ) {
-        this.observe = jest.fn();
-        this.disconnect = jest.fn();
-        this.trigger = () => {
-          callback();
-        };
-      });
-      // casted to any here as we don't really care about the interface we are mocking
-      // eslint-disable-next-line @typescript-eslint/no-explicit-any
-      (global as any).MutationObserver = mutationObserverMock;
-
+    it("should trigger the MutationObserver", async () => {
```

Comment:
I'm not entirely sure what this is testing now, but it certainly doesn't have anything to do with MutationObserver any more! I'm OK with a decision to change what this is testing (MutationObserver should be just an implementation detail anyway and we should test that the trap still works even as the focusable elements change, which I think is what we're doing now and seens perfectly correct) - but the name of the test should change to reflect this.

Reply:
Good point the description no longer matches what is being tested